From e1264462dca57b3c02a4a7131c356b0c33eed89a Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <Laurentiu.Tudor@freescale.com>
Date: Thu, 23 May 2013 12:21:51 +0300
Subject: [PATCH 374/430] powerpc/fsl: Introduce a generic pci-e compatible
 named "fsl,qoriq-pcie"

This will allow future code to simply check the pci-e controller
version register plus getting rid of the redundant version from
within the compatible string.

Signed-off-by: Laurentiu Tudor <Laurentiu.Tudor@freescale.com>
Change-Id: I7211a7c426f9e11920369355900ccd236fc248a1
Reviewed-on: http://git.am.freescale.net:8181/2670
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Wood Scott-B07421 <scottwood@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
[Original patch is from QorIQ-SDK-V1.4-20130625-yocto, just a minor
modification to port 3.4 kernel]
Signed-off-by: Wei Yang <Wei.Yang@windriver.com>
---
 arch/powerpc/boot/dts/fsl/b4420si-post.dtsi   |    2 +-
 arch/powerpc/boot/dts/fsl/b4860si-post.dtsi   |    2 +-
 arch/powerpc/boot/dts/fsl/bsc9132si-post.dtsi |    2 +-
 arch/powerpc/boot/dts/fsl/p1010si-post.dtsi   |    4 ++--
 arch/powerpc/boot/dts/fsl/p1023si-post.dtsi   |    6 +++---
 arch/powerpc/boot/dts/fsl/p2041si-post.dtsi   |    6 +++---
 arch/powerpc/boot/dts/fsl/p3041si-post.dtsi   |    8 ++++----
 arch/powerpc/boot/dts/fsl/p4080si-post.dtsi   |    6 +++---
 arch/powerpc/boot/dts/fsl/p5020si-post.dtsi   |    8 ++++----
 arch/powerpc/boot/dts/fsl/p5040si-post.dtsi   |    6 +++---
 arch/powerpc/boot/dts/fsl/t4240si-post.dtsi   |    8 ++++----
 11 files changed, 29 insertions(+), 29 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
index b9e9fdd..6816961 100644
--- a/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4420si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,b4420-pcie", "fsl,qoriq-pcie-v2.4";
+	compatible = "fsl,b4420-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
index 98e292e..b27cb93 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,b4860-pcie", "fsl,qoriq-pcie-v2.4";
+	compatible = "fsl,b4860-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/bsc9132si-post.dtsi b/arch/powerpc/boot/dts/fsl/bsc9132si-post.dtsi
index 20962ab..ea50a5f 100644
--- a/arch/powerpc/boot/dts/fsl/bsc9132si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/bsc9132si-post.dtsi
@@ -42,7 +42,7 @@
 
 /* controller at 0xa000 */
 &pci0 {
-	compatible = "fsl,bsc9132-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,bsc9132-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p1010si-post.dtsi b/arch/powerpc/boot/dts/fsl/p1010si-post.dtsi
index e949c47..1804ed7 100644
--- a/arch/powerpc/boot/dts/fsl/p1010si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p1010si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x9000 */
 &pci0 {
-	compatible = "fsl,p1010-pcie", "fsl,qoriq-pcie-v2.3", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p1010-pcie", "fsl,qoriq-pcie-v2.3", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -69,7 +69,7 @@
 
 /* controller at 0xa000 */
 &pci1 {
-	compatible = "fsl,p1010-pcie", "fsl,qoriq-pcie-v2.3", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p1010-pcie", "fsl,qoriq-pcie-v2.3", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi b/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
index f1105bf..6b5ab3e 100644
--- a/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p1023si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0xa000 */
 &pci0 {
-	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -60,7 +60,7 @@
 
 /* controller at 0x9000 */
 &pci1 {
-	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -79,7 +79,7 @@
 
 /* controller at 0xb000 */
 &pci2 {
-	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p1023-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi b/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
index 5ec6956..35d2d3b 100644
--- a/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -70,7 +70,7 @@
 
 /* controller at 0x201000 */
 &pci1 {
-	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -99,7 +99,7 @@
 
 /* controller at 0x202000 */
 &pci2 {
-	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p2041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi b/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
index df251f2..2780ae9 100644
--- a/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
@@ -44,7 +44,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -73,7 +73,7 @@
 
 /* controller at 0x201000 */
 &pci1 {
-	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -102,7 +102,7 @@
 
 /* controller at 0x202000 */
 &pci2 {
-	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -131,7 +131,7 @@
 
 /* controller at 0x203000 */
 &pci3 {
-	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p3041-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p4080si-post.dtsi b/arch/powerpc/boot/dts/fsl/p4080si-post.dtsi
index f6ee645..9fc15a0 100644
--- a/arch/powerpc/boot/dts/fsl/p4080si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p4080si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,p4080-pcie";
+	compatible = "fsl,p4080-pcie", "fsl,qoriq-pcie-v2.1", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -70,7 +70,7 @@
 
 /* controller at 0x201000 */
 &pci1 {
-	compatible = "fsl,p4080-pcie";
+	compatible = "fsl,p4080-pcie", "fsl,qoriq-pcie-v2.1", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -99,7 +99,7 @@
 
 /* controller at 0x202000 */
 &pci2 {
-	compatible = "fsl,p4080-pcie";
+	compatible = "fsl,p4080-pcie", "fsl,qoriq-pcie-v2.1", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p5020si-post.dtsi b/arch/powerpc/boot/dts/fsl/p5020si-post.dtsi
index a839b64..0718b1d 100644
--- a/arch/powerpc/boot/dts/fsl/p5020si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p5020si-post.dtsi
@@ -44,7 +44,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -73,7 +73,7 @@
 
 /* controller at 0x201000 */
 &pci1 {
-	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -102,7 +102,7 @@
 
 /* controller at 0x202000 */
 &pci2 {
-	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -131,7 +131,7 @@
 
 /* controller at 0x203000 */
 &pci3 {
-	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5020-pcie", "fsl,qoriq-pcie-v2.2", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/p5040si-post.dtsi b/arch/powerpc/boot/dts/fsl/p5040si-post.dtsi
index a895089..be9b0c1 100644
--- a/arch/powerpc/boot/dts/fsl/p5040si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p5040si-post.dtsi
@@ -41,7 +41,7 @@
 
 /* controller at 0x200000 */
 &pci0 {
-	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -69,7 +69,7 @@
 
 /* controller at 0x201000 */
 &pci1 {
-	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -97,7 +97,7 @@
 
 /* controller at 0x202000 */
 &pci2 {
-	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie-v2.2";
+	compatible = "fsl,p5040-pcie", "fsl,qoriq-pcie-v2.4", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
diff --git a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
index 8e25dfc..7536b59 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
@@ -53,7 +53,7 @@
 
 /* controller at 0x240000 */
 &pci0 {
-	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0";
+	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -80,7 +80,7 @@
 
 /* controller at 0x250000 */
 &pci1 {
-	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0";
+	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -107,7 +107,7 @@
 
 /* controller at 0x260000 */
 &pci2 {
-	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0";
+	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
@@ -134,7 +134,7 @@
 
 /* controller at 0x270000 */
 &pci3 {
-	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0";
+	compatible = "fsl,t4240-pcie", "fsl,qoriq-pcie-v3.0", "fsl,qoriq-pcie";
 	device_type = "pci";
 	#size-cells = <2>;
 	#address-cells = <3>;
-- 
1.7.5.4

